From 7f57cb31d58be161cfe4b92e8da5d81497adce85 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Enrique=20Oca=C3=B1a=20Gonz=C3=A1lez?= <eocanha@igalia.com>
Date: Tue, 26 Feb 2019 17:41:34 +0000
Subject: [PATCH 09/10] [BCM][GStreamer] Also try to match video decoder by
 type name

The video decoder on recent versions of the BCM_NEXUS GStreamer plugins
directly subclasses GstElement instead of GstVideoDecoder. The only way to
find the video decoder in that case is to try to match it by type name.

Without this patch, the video decoder isn't found and the current media time
is instead queried to the pipeline, giving incorrect values on BCM_NEXUS.
---
 .../WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
index 0c9bf6a6c95..26dbb0fda98 100644
--- a/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp
@@ -339,7 +339,7 @@ static GstElement* findVideoDecoder(GstElement *element)
         }
         g_value_unset (&item);
         gst_iterator_free(it);
-    } else if (GST_IS_VIDEO_DECODER(element))
+    } else if (GST_IS_VIDEO_DECODER(element) || g_str_has_suffix(G_OBJECT_TYPE_NAME(G_OBJECT(element)), "VideoDecoder"))
         re = element;
     return re;
 }
-- 
2.11.0

